components/calendar/src/calendar.rs

```diff
@@ -13,7 +14,7 @@ use crate::{Date, DateDuration, DateDurationUnit, Iso};
 /// allowing for direct construction, etc.
 pub trait Calendar {
     /// The internal type used to represent dates
-    type DateInner: PartialEq + Eq + Clone;
+    type DateInner: PartialEq + Eq + Clone + fmt::Debug;
```

Comment:
Nit: Why do we need all these bounds directly on the associated type, rather than applying the bounds only when we need them for a specific impl?

Reply:
because then`#[derive(Debug)]` (etc) stop working on types containing Dates. The custom derive cannot look up the types to figure out what the constraints should be -- it doesn't know that `Date<A>` internally contains `A::Calendar::DateInner`

Reply:
Can we `impl Debug for Date<A> where A::Calendar::DateInner: Debug { ... }` and then would the derive macro start working?

Reply:
Nope. The macro expects `where A: Debug`